dlib/dnn/loss.h

```diff
@@ -1294,6 +1294,113 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    class loss_mean_squared_
+    {
+    public:
+
+        typedef float training_label_type;
+        typedef float output_label_type;
+
+        template <
+            typename SUB_TYPE,
+            typename label_iterator
+            >
+            void to_label (
+                const tensor& input_tensor,
+                const SUB_TYPE& sub,
+                label_iterator iter
+                ) const
+            {
+                DLIB_CASSERT(sub.sample_expansion_factor() == 1);
+
+                const tensor& output_tensor = sub.get_output();
+
+                DLIB_CASSERT(output_tensor.nr() == 1 &&
+                             output_tensor.nc() == 1 &&
+                             output_tensor.k() == 1);
+                DLIB_CASSERT(input_tensor.num_samples() == output_tensor.num_samples());
+
+                const float* out_data = output_tensor.host();
+                for (long i = 0; i < output_tensor.num_samples(); ++i)
+                {
+                    *iter++ = out_data[i];
+                }
+            }
```

Comment:
hmm, looks like my emacs settings indents templates in a particular way. I have manually adapted to dlib style. Hopefully I fixed all of them :)